In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv('data/m_clean.csv')
df

distance  consume  speed  temp_inside  temp_outside gas_type  ac  rain  \
0        28.0      5.0     26         21.5            12      E10   0     0   
1        12.0      4.2     30         21.5            13      E10   0     0   
2        11.2      5.5     38         21.5            15      E10   0     0   
3        12.9      3.9     36         21.5            14      E10   0     0   
4        18.5      4.5     46         21.5            15      E10   0     0   
..        ...      ...    ...          ...           ...      ...  ..   ...   
371      16.0      3.7     39         24.5            18     SP98   0     0   
372      16.1      4.3     38         25.0            31     SP98   1     0   
373      16.0      3.8     45         25.0            19     SP98   0     0   
374      15.4      4.6     42         25.0            31     SP98   1     0   
375      14.7      5.0     25         25.0            30     SP98   1     0   

     sun  refill_liters    refill_gas  
0      0           45.0           E10  
1      0            0.0  not refilled  
2      0            0.0  not refilled  
3      0            0.0  not refilled  
4      0            0.0  not refilled  
..   ...            ...           ...  
371    0            0.0  not refilled  
372    0            0.0  not refilled  
373    0            0.0  not refilled  
374    0            0.0  not refilled  
375    0            0.0  not refilled  

[376 rows x 11 columns]

In [3]:
df['ratio'] = df['consume']/df['distance']
df.head()

distance  consume  speed  temp_inside  temp_outside gas_type  ac  rain  \
0      28.0      5.0     26         21.5            12      E10   0     0   
1      12.0      4.2     30         21.5            13      E10   0     0   
2      11.2      5.5     38         21.5            15      E10   0     0   
3      12.9      3.9     36         21.5            14      E10   0     0   
4      18.5      4.5     46         21.5            15      E10   0     0   

   sun  refill_liters    refill_gas     ratio  
0    0           45.0           E10  0.178571  
1    0            0.0  not refilled  0.350000  
2    0            0.0  not refilled  0.491071  
3    0            0.0  not refilled  0.302326  
4    0            0.0  not refilled  0.243243

In [4]:
e10 = df[df['gas_type'] == 'E10']
e10

distance  consume  speed  temp_inside  temp_outside gas_type  ac  rain  \
0        28.0      5.0     26         21.5            12      E10   0     0   
1        12.0      4.2     30         21.5            13      E10   0     0   
2        11.2      5.5     38         21.5            15      E10   0     0   
3        12.9      3.9     36         21.5            14      E10   0     0   
4        18.5      4.5     46         21.5            15      E10   0     0   
..        ...      ...    ...          ...           ...      ...  ..   ...   
332      16.1      5.1     30         25.0            16      E10   1     1   
333      16.0      4.0     42         22.0            15      E10   0     0   
334      15.4      4.8     40         22.0            18      E10   0     0   
335      17.2      3.9     35         22.0            16      E10   0     0   
336      16.6      4.6     50         22.0            25      E10   0     0   

     sun  refill_liters    refill_gas     ratio  
0      0           45.0           E10  0.178571  
1      0            0.0  not refilled  0.350000  
2      0            0.0  not refilled  0.491071  
3      0            0.0  not refilled  0.302326  
4      0            0.0  not refilled  0.243243  
..   ...            ...           ...       ...  
332    0            0.0  not refilled  0.316770  
333    0            0.0  not refilled  0.250000  
334    0            0.0  not refilled  0.311688  
335    0            0.0  not refilled  0.226744  
336    0            0.0  not refilled  0.277108  

[157 rows x 12 columns]

In [5]:
e10['ratio'].mean()

0.44074892012049055

In [13]:
# Your code here:
# One-sided test:
# H0: SP98 gas consumption is lower than E10 on average (<= 0.44)
# H1: SP98 gas consumption is higher than E10 on average (> 0.44)

In [8]:
SP98 = df[df['gas_type'] == 'SP98']
SP98.head()

distance  consume  speed  temp_inside  temp_outside gas_type  ac  rain  \
44       5.4      3.3     32         21.5             7     SP98   0     0   
45       2.0      9.9     21         21.5             4     SP98   0     0   
46      14.2      5.1     32         21.5             7     SP98   0     0   
47      16.0      4.9     25         21.5             8     SP98   0     0   
48      11.8      4.9     25         21.5            11     SP98   0     0   

    sun  refill_liters    refill_gas     ratio  
44    0           37.6          SP98  0.611111  
45    0            0.0  not refilled  4.950000  
46    0            0.0  not refilled  0.359155  
47    0            0.0  not refilled  0.306250  
48    0            0.0  not refilled  0.415254

In [9]:
from scipy.stats import ttest_1samp

In [11]:
stat, pval = ttest_1samp(SP98['ratio'], 0.44)

print('stat is  ', stat)
print('P-value for the one-tailed test is ', pval/2)

stat is   1.1931999135585298
P-value for the one-tailed test is  0.117044022783461


**Answer:**   
The P-value is less than 0.05, which means we reject $H_0:$  (SP98 gas consumption is lower than E10 on average <= 0.44)

In [14]:
import scipy.stats

confidence_level = 0.95
degrees_freedom = len(SP98['ratio']) - 1  
sample_mean = np.mean(SP98['ratio'])

sample_standard_error = scipy.stats.sem(SP98['ratio'])

confidence_interval = scipy.stats.t.interval(confidence_level, 
                                             degrees_freedom, 
                                             sample_mean, 
                                             sample_standard_error)

print( 'Confidence interval is ', confidence_interval, '.' )

Confidence interval is  (0.39473912050433846, 0.6241444658406701) .


Based on our confidence interval values, the SP98 gas consumption is between 0.39 and 0.62. 